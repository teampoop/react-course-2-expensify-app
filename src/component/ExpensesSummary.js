import React from 'react';
import { connect } from 'react-redux';

import numeral from 'numeral';

import selectExpenses from '../selectors/expenses';
import expensesTotal from '../selectors/expenses-total';

export class ExpensesSummary extends React.Component {

  expensesCount() {
    const count = this.props.expenses.length;
    return count === 1 ? `${count} expense` : `${count} expenses`;
  }

  expensesTotal() {
    return numeral(expensesTotal(this.props.expenses) / 100).format('$0,0.00');
  }

  render() {
    return (
      <p>Viewing {this.expensesCount()} totalling {this.expensesTotal()}</p>
    );
  };
}

const mapStateToProps = (state) => ({
  expenses: selectExpenses(state.expenses, state.filters)
});

export default connect(mapStateToProps,)(ExpensesSummary);