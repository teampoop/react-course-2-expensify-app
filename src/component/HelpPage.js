import React from 'react';

const HelpPage = () => (
  <div>
    This is from my Help Expense Component.
  </div>
);
export default HelpPage;