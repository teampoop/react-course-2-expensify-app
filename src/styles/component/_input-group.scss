.input-group {
  align-items: center;
  display: flex;
  margin-bottom: $m-size;
  flex-direction: column;
  @media (min-width: $desktop-breakpoint) {
    flex-direction: row;
    margin-botto: $l-size;
  }
}

.input-group__item {
  margin-bottom: $s-size;
  @media (min-width: $desktop-breakpoint) {
    margin-bottom: 0;
    margin-right: $s-size;
  }
}