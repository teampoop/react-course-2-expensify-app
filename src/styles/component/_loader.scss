.loader {
  height: 100vh;
  align-items: center;
  justify-content: center;
  width: 100vw;
  display: flex;
}

.loader__image {
  height: 6rem;
  width: 6rem;
}