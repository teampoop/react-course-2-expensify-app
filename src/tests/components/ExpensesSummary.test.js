import React from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
import { ExpensesSummary } from '../../component/ExpensesSummary';

test('should show two expenses totaling $1,096.95', () => {
  const e = [expenses[0], expenses[1]];
  const wrapper = shallow(<ExpensesSummary expenses={e} />);
  expect(wrapper).toMatchSnapshot();
});

test('should show 1 expense totalling $1.95', () => {
  const e = [expenses[0]];
  const wrapper = shallow(<ExpensesSummary expenses={e} />);
  expect(wrapper).toMatchSnapshot();
});