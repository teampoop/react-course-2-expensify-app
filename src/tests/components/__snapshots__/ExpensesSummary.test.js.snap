// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show 1 expense totalling $1.95 1`] = `
<p>
  Viewing 
  1 expense
   totalling 
  $1.95
</p>
`;

exports[`should show two expenses totaling $1,096.95 1`] = `
<p>
  Viewing 
  2 expenses
   totalling 
  $1,096.95
</p>
`;
